cmd/abapEnvironmentAssembly.go

```diff
@@ -0,0 +1,598 @@
+package cmd
+
+import (
+	"bytes"
+	"path"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http/cookiejar"
+
+	"github.com/SAP/jenkins-library/pkg/abaputils"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+func abapEnvironmentAssembly(config abapEnvironmentAssemblyOptions, telemetryData *telemetry.CustomData, commonPipelineEnvironment *abapEnvironmentAssemblyCommonPipelineEnvironment) {
+	// for command execution use Command
+	c := command.Command{}
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	// for http calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
```

Comment:
It would be better to initialize the abapUtils structure here and pass it to the functions. You can use the interface abaputils.Communication in the signature.

With this approach it is easier to mock the dependencies (for unit tests) if necessary.

Reply:
This is also true for the http client